src/Query/Builder.php

```diff
@@ -1057,40 +1061,43 @@ protected function compileWhereBasic(array $where): array
     {
         extract($where);
 
-        // Replace like or not like with a Regex instance.
+        // Replace like with a Regex instance.
         if (in_array($operator, ['like', 'not like'])) {
-            if ($operator === 'not like') {
-                $operator = 'not';
-            } else {
+            if ($operator === 'like') {
                 $operator = '=';
+            } else {
+                $operator = 'not';
             }
 
-            // Convert to regular expression.
-            $regex = preg_replace('#(^|[^\\\])%#', '$1.*', preg_quote($value));
-
-            // Convert like to regular expression.
-            if (! Str::startsWith($value, '%')) {
-                $regex = '^'.$regex;
-            }
-            if (! Str::endsWith($value, '%')) {
-                $regex .= '$';
-            }
-
-            $value = new Regex($regex, 'i');
-        } // Manipulate regexp operations.
-        elseif (in_array($operator, ['regexp', 'not regexp', 'regex', 'not regex'])) {
+            // Convert % and _ to regex, and unescape \% and \_
+            $regex = preg_replace(
+                ['#(^|[^\\\])%#', '#(^|[^\\\])_#', '#\\\\\\\(%|_)#'],
+                ['$1.*', '$1.', '$1'],
+                preg_quote($value),
+            );
+            $value = new Regex('^'.$regex.'$', 'i');
+        } // Manipulate regex operations.
+        elseif (in_array($operator, ['regex', 'not regex'])) {
             // Automatically convert regular expression strings to Regex objects.
-            if (! $value instanceof Regex) {
-                $e = explode('/', $value);
-                $flag = end($e);
-                $regstr = substr($value, 1, -(strlen($flag) + 1));
-                $value = new Regex($regstr, $flag);
+            if (is_string($value)) {
+                $delimiter = substr($value, 0, 1);
+                if (! in_array($delimiter, ['/', '#', '~'])) {
+                    throw new \InvalidArgumentException(sprintf('Regular expressions must be surrounded by delimiter "/". Got "%s"', $value));
```

Comment:
In addition to updating this to report the list of permitted delimiters, Consider printing `$delimiter` instead of `$value` for a more concise error message. In this case, we have enough information to inform the user that the first character is unexpected.

Reply:
We provide more information by providing the full regex. And usually that's not very long.